### PR TITLE
Improve malformed RegExp named groups coverage

### PR DESCRIPTION
**Pull request by shvaikalesh**
This change:
a) Moves named references with look-behind tests to separate file, allowing engines without look-behind assertions support (JSC) to run them.
b) Adds a few tests with both malformed named and **numeric** references, as they both require reparsing and may interfere with each other.

JSC bug: [Invalid numeric and named references should be early syntax errors](https://bugs.webkit.org/show_bug.cgi?id=178175).
//cc @mathiasbynens

### CODE REVIEW

**Comment by rwaldron** [comment]
@mathiasbynens _gentle nudge_

File: test/annexB/built-ins/RegExp/named-groups/non-unicode-malformed.js [hunk 1]
```diff
@@ -17,11 +17,9 @@ assert(/\k/.test("k"));
...
+assert.compareArray(["k<a>"], "xxxk<a>xxx".match(/\k<a>/));
+assert.compareArray(["k<a"], "xxxk<a>xxx".match(/\k<a/));
 
-// A couple of corner cases around '\k' as named back-references vs. identity
-// escapes.
-assert(/\k<a>(?<=>)a/.test("k<a>a"));
-assert(/\k<a>(?<!a)a/.test("k<a>a"));
 assert(/\k<a>(<a>x)/.test("k<a><a>x"));
+assert(/\k<a>\1/.test("k<a>\x01"));
+assert(/\1(b)\k<a>/.test("bk<a>"));
```
Comment (leobalter) [inline test/annexB/built-ins/RegExp/named-groups/non-unicode-malformed.js:25]: while the assertions are fine, it seems it would be slightly better to capture the results if they were positioned in a new test file, for similar reasons some assertions here are being set in a new file.
